Add cc pipeline properties (app-name specifically) to ALM

brendankellyibm opened the pull request:
### Description

Added `cc_pipeline_properties` to standard/basic definitions of the ALM component, so that we can have the `app-name `parameter available in the CC toolchain for dynamic scans

### Release required?
<!--- Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning). --->

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

<!--- If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module. --->

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

in-1911 commented on solutions/standard/stack_definition.json:
```diff
@@ -701,6 +701,10 @@
         },
+        {
+          "name": "cc_pipeline_properties",
+          "value": "[{\"app-name\": \"rag-sample-app\"}]"
```
I think it should follow the schema like
```json
[
  {
    "pipeline_id": "cc",
    "properties": [
      {
        "name": "app-name",
        "type": "text",
        "value": "...."
      }
...
```

brendankellyibm replied:
You're right - will make the change

brendankellyibm replied:
It should follow the right schema now, I wrongly assumed it was using the same approach as DA-defined variables.

brendankellyibm commented:
I'll make more changes to this - will need to include the existing default variables.

brendankellyibm commented:
Should be ok now.

- I added the default cc pipeline variables from the stack.
- Changed: `opt-in-dynamic-scan`, `opt-in-dynamic-ui-scan` to enable scans from CC.
- Added: `app-name`

brendankellyibm commented:
/run pipeline

in-1911 commented on solutions/standard/stack_definition.json:
```diff
@@ -701,6 +701,10 @@
         },
+        {
+          "name": "cc_pipeline_properties",
+          "value": "[{\"pipeline_id\":\"cc\",\"properties\":[{\"name\":\"app-name\",\"type\":\"text\",\"value\":\"rag-sample-app\"},{\"name\":\"cos-api-key\",\"type\":\"secure\",\"value\":\"\"},{\"name\":\"cos-bucket-name\",\"type\":\"text\",\"value\":\"\"},{\"name\":\"cos-endpoint\",\"type\":\"text\",\"value\":\"\"},{\"name\":\"cra-bom-generate\",\"type\":\"single_select\",\"value\":\"1\",\"enum ...
```
I think we may need to set pipeline-config-branch to "main" - we are seeing some issues with CD pipeline on ROKS deployment.

brendankellyibm replied:
Updated with that value to see if it fixes your clone issue - also added entire CD pipeline parameters for ROKS, that was missing.

ocofaigh commented:
Now that https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/pull/189 is merged, we need to bump the sample app to the latest version in this PR before merge, and then use this PR to generate 2.0.1.

We also probably want to do an upgrade test from 2.0.0 first too.

ocofaigh commented:
/run pipeline

ocofaigh commented:
Saw this error:

```
         2024/10/14 14:24:26 Terraform init | Error while installing ibm-cloud/ibm v1.69.2: could not query provider
         2024/10/14 14:24:26 Terraform init | registry for registry.terraform.io/ibm-cloud/ibm: failed to retrieve
         2024/10/14 14:24:26 Terraform init | cryptographic signature for provider: the request failed after 2 attempts,
         2024/10/14 14:24:26 Terraform init | please try again later: 500 Internal Server Error returned from github.com
         2024/10/14 14:24:26 Terraform INIT error: Terraform INIT errorexit status 1
         2024/10/14 14:24:26 Could not execute job: Error : Terraform INIT errorexit status 1
```

And an issue deleting resource group (which I have a support case open for). Unrelated to the changes in this pr, so re-running..

ocofaigh commented:
/run pipeline

ocofaigh commented:
There is also a fix for cloud logs which I want to get into 2.0.1 -> https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/issues/587

ocofaigh commented:
Ah I just realised that if we pick up that fix, we will have to remove logdna support, as it has been removed from the module now. I think we will have to proceed with what is in this PR for the 2.0.1 release.
The Cloud logs issue can be solved by simply re-running the apply (since its a timing issue with auth policy). I'll document that for now.

ocofaigh commented:
On second thoughts, the RAG stack no longer deploys logdna, so its probably fine - waiting for Igor to confirm

in-1911 commented:
As discussed on slack - I am ok with bringing in the Observability DA that no longer has LogDNA option at all - as long as RAG Stack 2.0.0 is still available in case I will have troubles with SCC scans when LogDNA is not deployed.

I think the effects of the Cloud Logs issue are more severe than just a failed deployment that can be retried - last time my Cloud Log instance got stuck in "provisioning" status  for more than a day and caused the terraform resource in the state to be in a "deposed" status which prevented a normal redeployment.

ocofaigh commented:
/run pipeline

ocofaigh commented:
Need to force merge this PR - we can't get the test wrapper to pass due to know issues with projects reporting the incorrect state for a period of time. Validation will occur in catalog anyway.

terraform-ibm-modules-ops commented:
:tada: This PR is included in version 2.0.1 :tada:

The release is available on:
- [GitHub release](https://github.com/terraform-ibm-modules/stack-retrieval-augmented-generation/releases/tag/v2.0.1)
- `v2.0.1`

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
